modules/gateway/peers.go

```diff
@@ -304,6 +325,19 @@ func acceptableVersion(version string) error {
 	return nil
 }
 
+// acceptableSessionHeader returns an error
+// if the remote header indicates that a connection is impossible.
+func acceptableSessionHeader(ours, theirs *sessionHeader) error {
```

Comment:
don't pass `sessionHeader` by pointer. It's small enough that there's no reason to worry about copying it around.